rajawali/src/main/java/c/org/rajawali3d/sceneview/SceneView.java

```diff
@@ -0,0 +1,227 @@
+package c.org.rajawali3d.sceneview;
+
+import android.graphics.Rect;
+import android.opengl.GLES20;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import c.org.rajawali3d.annotations.GLThread;
+import c.org.rajawali3d.annotations.RequiresReadLock;
+import c.org.rajawali3d.camera.Camera;
+import c.org.rajawali3d.engine.Engine;
+import c.org.rajawali3d.engine.RenderModel;
+import c.org.rajawali3d.engine.RenderView;
+import c.org.rajawali3d.object.renderers.ObjectRenderer;
+import c.org.rajawali3d.scene.Scene;
+import c.org.rajawali3d.scene.graph.NodeMember;
+import java.util.List;
+import java.util.concurrent.locks.Lock;
+import net.jcip.annotations.GuardedBy;
+import org.rajawali3d.math.Matrix4;
+import org.rajawali3d.util.RajLog;
+
+/**
+ * Author: Randy Picolet
+ */
+
+public class SceneView implements RenderView {
+    private static final String TAG = "SceneView";
+
+    @Nullable
+    private Engine engine;
+
+    @NonNull
+    private Scene scene;
+
+    // TODO probably need a decorator Viewport class around the Rect to flag changes...
+    @NonNull
+    private Rect viewport;
+
+    /**
+     * The {@link Camera} currently being used to render the scene.
+     */
+    @GuardedBy("nextCameraLock")
+    private Camera currentCamera;
+
+    private Camera nextCamera; // The camera the scene should switch to on the next frame.
+    private final Object nextCameraLock = new Object(); // Camera switching lock
+
+    @NonNull
+    @GLThread
+    private Matrix4 viewProjectionMatrix = new Matrix4();
+
+    @Nullable
+    @GLThread
+    private ObjectRenderer lastUsedRenderer; // Reference to the last used object engine
```

Comment:
Makes sense, since otherwise every SceneView.onRenderView() would have to reset the whole state each frame.  Maybe just pass it in and return it both?